### PR TITLE
Document Default Error Responses

### PR DESCRIPTION
**Pull request by muya**
Include documentation for 500 & 503 responses in default schema definition

This ensures that they get included in the Open API documentation generated.

- chore: update tests to match updated responses

#### Demo / Screenshots

|Before|After|
|---|---|
|screenshot-20230710_140757|under-pressure-after-adding-503-docs-resized |

#### Notes to Reviewer(s)

- Addition of docs for other responses make for more complete API documentation when using the healthcheck feature
- Open to suggestions on wording of descriptions
- Try out the updated documentation version here: [muya/fastify-under-pressure-update-demo](https://github.com/muya/fastify-under-pressure-update-demo)

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

File: index.js [hunk 1]
```diff
@@ -107,10 +107,29 @@ async function fastifyUnderPressure (fastify, opts) {
...
+            }
+          },
+          503: {
+            type: 'object',
+            description: 'Health Check Failed',
+            properties: {
+              code: { type: 'string', description: 'Error code associated with the failing check', example: 'FST_UNDER_PRESSURE' },
+              error: { type: 'string', description: 'Error thrown during health check', example: 'Service Unavailable' },
+              message: { type: 'string', description: 'Error message to explain health check failure', example: 'Service Unavailable' },
+              statusCode: { type: 'number', description: 'Code representing the error. Currently matches the HTTP response code.', example: 503 }
```
Comment (Uzlopak) [inline index.js:131]: this seems odd. statuscode will always be HTTP respone code.

**Review by Uzlopak — CHANGES_REQUESTED** [review]
I kind of doubt that this so that much correct. It is good to add the 500 and 503 status codes to the schema. But the errors are thrown so the default error handler takes as we did not specify setErrorHandler for this plugin. 

So if somebody changes the default errorHandler than the schema is not correct as the error handler will be generating a different response.

So we have to add an error Handler with setErrorHandler for this plugin, to ensure that the api contract is ensured even if the defaultErrorHandler is changed.

**Comment by Uzlopak** [comment]
@Eomm 
Do you concur?

**Comment by muya** [comment]
> I kind of doubt that this so that much correct. It is good to add the 500 and 503 status codes to the schema. But the errors are thrown so the default error handler takes as we did not specify setErrorHandler for this plugin.
> 
> So if somebody changes the default errorHandler than the schema is not correct as the error handler will be generating a different response.
> 
> So we have to add an error Handler with setErrorHandler for this plugin, to ensure that the api contract is ensured even if the defaultErrorHandler is changed.

Hi @Uzlopak, thank you for taking a look.

I agree that this implementation only allows for the default responses.

However, I think that this is okay, since an API spec is meant to define the "schema", and not the actual response that will be returned. 
For example, if an API returns 10 different variations of a message, then we don't have to document all the 10 responses, we only have to define the "structure".

Since there's currently no support for defining a custom error handler, the API is always returning the same payload in case of 500 & 503 errors.

If we decide to add the error handler, we would then also update the schema definition to accept a custom payload (i.e. in the same way we're combining the custom success response in the `200 OK` response).

What do you think?

File: index.js (hunk 1 above)
Comment (muya) [reply index.js]: @Uzlopak This is updated ✅

**Comment by Uzlopak** [comment]
But afaik specifiying a plugin specific errorhandler is possible by using the setErrorHandler method. So we could specify a custom errorhandler and uphold the api contract even if the default error handler is modified.

**Comment by Uzlopak** [comment]
Oh, I get it now. If you define another error handler, then the dev should specify the schema by himself?

**Comment by mcollina** [comment]
If what is in this pr is ok, let's land it and then you can open a fresh issue with the fix.

**Review by Uzlopak — APPROVED** [review]

**Pull request merged**